stream/src/placeholder/utils/nonce.rs

```diff
@@ -1,15 +1,51 @@
+use crate::placeholder::Error;
 use chacha20poly1305::Nonce;
 
-pub fn encode(dialer: bool, iter: u16, seq: u64) -> Nonce {
-    let mut result = Nonce::default();
-    if dialer {
-        result[0] = 0b10000000; // Set the first bit of the byte
+/// A struct that holds the nonce information.
+/// Dialer is a boolean that is true if the nonce is for the dialer side.
+pub struct Info {
+    dialer: bool,
+    iter: u16,
+    seq: u64,
+}
+
+impl Info {
+    pub fn new(dialer: bool) -> Self {
+        Self {
+            dialer,
+            iter: 0,
+            seq: 0,
+        }
+    }
+
+    /// Increments the nonce sequence number.
+    /// If the sequence number overflows, the iteration number is incremented and the sequence number is reset to 0.
```

Comment:
nit: I usually put a `\n` between the "simple description" and "extended notes".